CLN: fix mypy error pandas/tests/plotting/test_backend.py

saska opened the pull request:
GH28926, mypy didn't like setting non-existant attribute on module.
Dummy function, circumvented with `__setattr__`

- [x] addresses part of #28926
- [x] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

saska committed:
CLN: fix mypy error pandas/tests/plotting/test_backend.py

GH28926, mypy didn't like setting non-existant attribute on module.
Dummy function, circumvented with __setattr__

simonjayhawkins commented on pandas/tests/plotting/test_backend.py:
```diff
@@ -9,7 +9,7 @@
 
 dummy_backend = types.ModuleType("pandas_dummy_backend")
-dummy_backend.plot = lambda *args, **kwargs: None
+dummy_backend.__setattr__("plot", lambda *args, **kwargs: None)
```
use setattr() (or a # type: ignore) see https://github.com/python/mypy/issues/7171#issuecomment-520407743

saskakarsi replied:
Done! Personally preferred the setattr() way over ignoring a test line here, comment didn't specify why monkey-patching would be bad here - can do it the other way as well if you prefer and we just want to skip the line entirely.

simonjayhawkins replied:
Thanks for the update, but can you review the merge. looks like something went awry.

saskakarsi replied:
Oh jeez. Will do!

saskakarsi replied:
Alright, Travis CI wanted to run a couple of times but merge should be fine now!

simonjayhawkins reviewed (changes requested):
@saskakarsi Thanks for the PR. minor change, also can you merge master

saska committed:
Amended #29016 according to comments

Switched from obj.__setattr__(...) to setattr(obj, ...)
https://github.com/python/mypy/issues/7171#issuecomment-520407743

saska committed:
Resolve setup.cfg merge conflict

saska committed:
Fix broken merge

commit c7f03eb7cef276f9e5251ca2bcfee75b03751cbd

simonjayhawkins reviewed (approved):
@saskakarsi lgtm @WillAyd

WillAyd commented:
@TomAugspurger mind taking a look at this one?

TomAugspurger commented:
Thanks!
